src/ui/flutter_app/lib/mill/position.dart

```diff
@@ -467,159 +467,141 @@ class Position {
       return false;
     }
 
-    if (phase == Phase.ready) {
-      start();
-    }
+    if (phase == Phase.ready) start();
 
-    // TODO: use switch case
-    if (phase == Phase.placing) {
-      piece = sideToMove;
-      if (pieceInHandCount[us] != null) {
-        pieceInHandCount[us] = pieceInHandCount[us]! - 1;
-      }
+    switch (phase) {
+      case Phase.placing:
+        piece = sideToMove;
+        if (pieceInHandCount[us] != null) {
+          pieceInHandCount[us] = pieceInHandCount[us]! - 1;
+        }
 
-      if (pieceOnBoardCount[us] != null) {
-        pieceOnBoardCount[us] = pieceOnBoardCount[us]! + 1;
-      }
+        if (pieceOnBoardCount[us] != null) {
+          pieceOnBoardCount[us] = pieceOnBoardCount[us]! + 1;
+        }
 
-      _grid[squareToIndex[s]!] = piece;
-      board[s] = piece;
+        _grid[squareToIndex[s]!] = piece;
+        board[s] = piece;
 
-      record = "(${fileOf(s)},${rankOf(s)})";
+        record = "(${fileOf(s)},${rankOf(s)})";
 
-      updateKey(s);
+        updateKey(s);
 
-      currentSquare = s;
+        currentSquare = s;
 
-      final int n = millsCount(currentSquare);
+        final int n = millsCount(currentSquare);
 
-      if (n == 0) {
-        assert(
-          pieceInHandCount[PieceColor.white]! >= 0 &&
-              pieceInHandCount[PieceColor.black]! >= 0,
-        );
+        if (n == 0) {
+          assert(
+            pieceInHandCount[PieceColor.white]! >= 0 &&
+                pieceInHandCount[PieceColor.black]! >= 0,
+          );
 
-        if (pieceInHandCount[PieceColor.white] == 0 &&
-            pieceInHandCount[PieceColor.black] == 0) {
-          if (isGameOver()) {
-            return true;
-          }
+          if (pieceInHandCount[PieceColor.white] == 0 &&
+              pieceInHandCount[PieceColor.black] == 0) {
+            if (isGameOver()) return true;
```

Comment:
Why was it named `checkIfGameIsOver` instead of `isGameOver` at the time? It borrowed from the name in the morris program written by a Google Software Engineer - Cristian Patrasciuc, because this function performs complex calculations instead of simply reading the state, so it is named like this. But the impact is not big.

Reply:
see above (separate branch)